[WOOMOB-1029][Woo POS][Settings] Store details

kidinov opened the pull request:
WOOMOB-1029

### Description
Add store section to Woo POS settings with repository architecture improvements and comprehensive testing.

Key changes:
- Add store settings section showing store name, address, and email
- Display receipt information with store details and refund policy for WC 10+
- Split repository architecture into separate store and receipt data sources
- Implement proper loading states with shimmer placeholders
- Add comprehensive unit tests for ViewModel

### Testing information
- Store data loads from cached site settings for fast display
- Receipt data fetches from POS settings API endpoint for WC 10+ only
- Loading states properly handle different scenarios (loading, error, not supported)
- Unit tests cover all state transitions and error conditions

https://github.com/user-attachments/assets/2621d6c5-f765-44d2-8ef9-43550cc523b5

kidinov committed:
Create store settings package with ViewModel and Screen

kidinov committed:
Refactor WooPOS store settings with sealed classes and repository pattern

kidinov committed:
Improve categories layout with fixed positioning and type safety

kidinov committed:
Redesign store settings screen following POS menu item pattern

kidinov committed:
Set STORE as default selected category in settings

kidinov committed:
Increase start padding for section titles

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 300 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

kidinov committed:
Refactor store settings to split data sources and add POS receipt endpoint

- Split WooPosStoreReceiptRepository into separate store and receipt repositories
- Add POS settings endpoint to WooSystemRestClient for WC 10+ receipt data
- Update state management to handle separate loading states for store and receipt
- Store info loads immediately, receipt section shows loading indicator separately
- Receipt data fetched only on WooCommerce 10+ via new API endpoint
- Improved error handling with separate states for each data source

kidinov committed:
Refactor POS settings state management and update API endpoint

kidinov committed:
Add loading state for store information and update store info structure

kidinov committed:
Refactor store settings loading sections and improve UI structure

kidinov committed:
Rename store and receipt repository classes to improve clarity and consistency

kidinov committed:
Added a test

kidinov committed:
Add comprehensive unit tests for WooPos store and receipt repositories with real Settings objects

kidinov committed:
Remove unrelated files from git tracking

samiuelson commented on WooCommerce/src/main/res/values/strings.xml:
```diff
@@ -4547,6 +4547,15 @@
+    <string name="woopos_settings_store_phone_label">Phone</string>
+    <string name="woopos_settings_store_email_label">Email</string>
+    <string name="woopos_settings_refund_policy_label">Refund &amp; Returns Policy</string>
+    <string name="woopos_settings_store_not_set">Not set</string>
```
💡 Let's move new strings with `woopos_settings_` prefix so that they are not in the middle of `woopos_scanning_` strings

kidinov replied:
Done 60a160509f15c1d71532a8d97f9a12460c8faa45

samiuelson commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/settings/details/store/WooPosSettingsReceiptRepository.kt:
```diff
@@ -0,0 +1,59 @@
+        }
+
+    private fun isWooCommerceVersionAtLeast10(): Boolean {
+        val wooCoreVersion = getWooCoreVersion() ?: return false
```
💡 Consider making a reusable use case class, because we have duplicate logic in `WooPosEmailReceiptRepository::isWooCoreSupportsPOSReceipts`

kidinov replied:
I’m not sure if version comparison is worth it since we already have the reusable “semverCompareTo” and version fetching `getWooCoreVersion`. In my opinion, it might actually make things more complex than simpler.

samiuelson commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/settings/details/store/WooPosSettingsStoreScreen.kt:
```diff
@@ -0,0 +1,263 @@
+        icon = Icons.Default.Store,
+        title = stringResource(R.string.woopos_settings_store_name_label),
+        subtitle = storeInfo.storeName.ifBlank { stringResource(R.string.woopos_settings_store_not_set) },
+        onClick = { }
```
❓ Are these items gonna be clickable? If not, let's remove ripple

kidinov replied:
Good point - changed here

5db5be2bf22b5b34bea6d4e7bc3930eb5d077ac4

samiuelson reviewed (approved):
Overall, LGTM! Co-pilot and I left some comments, take a look :)

kidinov committed:
Improve store address formatting by adding conditional commas and spaces

kidinov committed:
Fixed formatting

kidinov commented on WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/settings/details/store/WooPosSettingsStoreRepositoryTest.kt:
```diff
@@ -0,0 +1,235 @@
+        val result = repository.getStoreInfo()
+
+        // THEN
+        assertThat(result.address).isEqualTo(", London, UK")
```
Fixed 5e8b0515f129df525088f79f829a1d1f29f24595

kidinov commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/settings/details/store/WooPosSettingsStoreRepository.kt:
```diff
@@ -0,0 +1,37 @@
+            settings?.city?.let { if (it.isNotBlank()) append(", $it") }
+            settings?.stateCode?.let { if (it.isNotBlank()) append(", $it") }
+            settings?.postalCode?.let { if (it.isNotBlank()) append(" $it") }
+            settings?.countryCode?.let { if (it.isNotBlank()) append(", $it") }
```
Fixed 5e8b0515f129df525088f79f829a1d1f29f24595

kidinov committed:
Better grouping of the strings

kidinov committed:
Refactor WooPosSettingsDetailsMenuItem to make onClick optional
